app/user.go

```diff
@@ -0,0 +1,597 @@
+)
+
+var (
+	userIDFlag = &cli.StringFlag{
+		Name:    userIDFlagName,
+		Usage:   "The user id",
+		Aliases: []string{"i"},
+	}
+	userEmailFlag = &cli.StringFlag{
+		Name:    userEmailFlagName,
+		Usage:   "The user email address of the user to be invited",
+		Aliases: []string{"e"},
+	}
+)
+
+type UserClient struct {
+	client authservice.AuthServiceClient
+	ctx    context.Context
+}
+
+type GetUserClientFn func(ctx *cli.Context) (*UserClient, error)
+
+func GetUserClient(ctx *cli.Context) (*UserClient, error) {
+	ct, conn, err := GetServerConnection(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return &UserClient{
+		client: authservice.NewAuthServiceClient(conn),
+		ctx:    ct,
+	}, nil
+}
+
+func (c *UserClient) listUsers(
+	namespace string,
+) error {
+	totalRes := &authservice.GetUsersResponse{}
+	pageToken := ""
+	for {
+		res, err := c.client.GetUsers(c.ctx, &authservice.GetUsersRequest{
+			PageToken: pageToken,
+			Namespace: namespace,
+		})
+		if err != nil {
+			return err
+		}
+		totalRes.Users = append(totalRes.Users, res.Users...)
+		// Check if we should continue paging
+		pageToken = res.NextPageToken
+		if len(pageToken) == 0 {
+			return PrintProto(totalRes)
+		}
+	}
+}
+
+func (c *UserClient) getUser(userID, userEmail string) (*auth.User, error) {
+	if userID == "" && userEmail == "" {
+		return nil, fmt.Errorf("both user-id and user-email not set")
+	}
+
+	if userID != "" && userEmail != "" {
+		return nil, fmt.Errorf("both user-id and user-email set")
+	}
+	res, err := c.client.GetUser(c.ctx, &authservice.GetUserRequest{
+		UserId:    userID,
+		UserEmail: userEmail,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if res.User == nil || res.User.Id == "" {
+		// this should never happen, the server should return an error when the user is not found
+		return nil, fmt.Errorf("invalid user returned by server")
+	}
+	return res.User, nil
+}
+
+func (c *UserClient) getUserAndRoles(userID, userEmail string) (*auth.User, []*auth.Role, error) {
+	user, err := c.getUser(userID, userEmail)
+	if err != nil {
+		return nil, nil, err
+	}
+	var pageToken string
+	var roles []*auth.Role
+	for {
+		res, err := c.client.GetRoles(c.ctx, &authservice.GetRolesRequest{
+			PageToken: pageToken,
+			UserId:    user.Id,
+		})
+		if err != nil {
+			return nil, nil, err
+		}
+		roles = append(roles, res.Roles...)
+		// Check if we should continue paging
+		pageToken = res.NextPageToken
+		if len(pageToken) == 0 {
+			break
+		}
+	}
+
+	return user, roles, nil
+}
+
+func (c *UserClient) inviteUsers(
+	ctx *cli.Context,
+	emails []string,
+	namespacePermissions []string,
+	accountRole string,
+) error {
+	if len(accountRole) == 0 {
+		return errors.New("account role required for inviting new users")
+	}
+
+	// the role ids to invite the users for
+	var roleIDs []string
+
+	// first get the required account role
+	role, err := getAccountRole(c.ctx, c.client, accountRole)
+	if err != nil {
+		return err
```

Comment:
is the error returned from this clear to someone trying to invite a user? Should we wrap the error message here with "invalid account role specified: %v" ?